src/Components/SceneList/Internal/SceneDialog.tsx

```diff
@@ -264,23 +292,29 @@ const SceneDialog: React.FC<ISceneDialogProps> = ({
                 selectedSceneCopy.assets[0].url = blobUrl;
                 setScene(selectedSceneCopy);
             } else {
-                setNewSceneBlobUrl(blobUrl);
+                dispatch({
+                    type: SET_SCENE_BLOB_URL,
+                    payload: blobUrl
+                });
             }
         },
         [sceneToEdit, scene]
     );
 
     const handleFileOverwriteChange = useCallback((_e, checked: boolean) => {
-        setIsOverwriteFile(checked);
+        dispatch({
+            type: SET_IS_OVER_WRITE_FILE,
+            payload: checked
+        });
     }, []);
 
     const getNewScene = (assetUrl: string) => {
         return {
```

Comment:
teh state should already have a constructed scene object, this should not be required. Please have the reducer return the built out state object directly.